src/main/java/com/telefonica/iot/tidoop/backends/ckan/CKANBackend.java

```diff
@@ -0,0 +1,270 @@
+     * @param ckanHost
+     * @param ckanPort
+     * @param ssl
+     * @param apiKey
+     */
+    public CKANBackend(String ckanHost, String ckanPort, boolean ssl, String apiKey) {
+        this.ckanHost = ckanHost;
+        this.ckanPort = ckanPort;
+        this.logger = Logger.getLogger(CKANBackend.class);
+        this.apiKey = apiKey;
+        this.ssl = ssl;
+        this.httpClientFactory = new HttpClientFactory(ssl);
+    } // CKANBackend
+    
+    /**
+     * Sets the Http client factory. It is protected since it is only going to be used by the tests.
+     * @param httpClientFactory
+     */
+    protected void setHttpClientFactory(HttpClientFactory httpClientFactory) {
+        this.httpClientFactory = httpClientFactory;
+    } // setHttpClientFactory
+    
+    /**
+     * Gets the package identifiers/names within a given organization id/name.
+     * @param orgId
+     * @return The package identifiers/names within the given organization id/name
+     */
+    public List<String> getPackages(String orgId) {
+        logger.debug("Getting the packages within " + orgId + " organization");
+
+        try {
+            List<String> resIds = new ArrayList<String>();
+            String url = "http" + (ssl ? "s" : "") + "://" + ckanHost + ":" + ckanPort
+                    + "/api/3/action/organization_show?id=" + orgId;
+            CKANResponse resp = doCKANRequest("GET", url, "");
+            JSONObject result = (JSONObject) resp.getJsonObject().get("result");
+            JSONArray pkgs = (JSONArray) result.get("packages");
+            
+            for (Object pkg1 : pkgs) {
+                JSONObject pkg = (JSONObject) pkg1;
+                resIds.add((String) pkg.get("id"));
+            } // for
+            
+            return resIds;
+        } catch (Exception e) {
+            return null;
+        } // try catch
+    } // getPackages
+    
+    /**
+     * Gets the resource identifiers/names within a given package id/name.
+     * @param pkgId
+     * @return The resource identifiers/names within the given package id/name
+     */
+    public List<String> getResources(String pkgId) {
+        logger.debug("Getting the resources within " + pkgId + " + package");
+        
+        try {
+            List<String> resIds = new ArrayList<String>();
+            String url = "http" + (ssl ? "s" : "") + "://" + ckanHost + ":" + ckanPort
+                    + "/api/3/action/package_show?id=" + pkgId;
+            CKANResponse resp = doCKANRequest("GET", url, "");
+            JSONObject result = (JSONObject) resp.getJsonObject().get("result");
+            JSONArray resources = (JSONArray) result.get("resources");
+            
+            for (Object res : resources) {
+                JSONObject resource = (JSONObject) res;
+                resIds.add((String) resource.get("id"));
+            } // for
+            
+            return resIds;
+        } catch (Exception e) {
+            return null;
```

Comment:
Wouldn't it be better to elevate the exception or including code to deal with the possibility of the value returned by getResources() to be null? According to https://github.com/telefonicaid/fiware-tidoop/blob/feature/1_ckan_extensions/src/main/java/com/telefonica/iot/tidoop/hadoop/ckan/CKANInputFormat.java#L218 no checking is done. At least some logging would be recommended, doesn't it?